lib/agent0/interactive_fuzz/fuzz_long_short_maturity_values.py

```diff
@@ -0,0 +1,191 @@
+    "bond_reserves",
+    "lp_total_supply",
+    "longs_outstanding",
+]
+
+
+# %%
+# Open some trades
+# TODO: include add liquidity
+trade_events: list[tuple[InteractiveHyperdriveAgent, OpenLong | OpenShort]] = []
+for trade in trade_list:
+    agent, trade_type, trade_amount = trade
+    if trade_type == HyperdriveActionType.OPEN_LONG:
+        trade_event = agent.open_long(base=trade_amount)
+    elif trade_type == HyperdriveActionType.OPEN_SHORT:
+        trade_event = agent.open_short(bonds=trade_amount)
+    else:
+        raise AssertionError(f"{trade_type=} is not supported.")
+    trade_events.append((agent, trade_event))
+
+# automatically created by sending transactions
+starting_checkpoint = interactive_hyperdrive.hyperdrive_interface.current_pool_state.checkpoint
+
+# %%
+# snapshot the chain, so we can load the snapshot & close in different orders
+chain.save_snapshot()
+
+check_data: dict | None = None
+# %%
+# advance the time to at least the position duration, maximum of two position durations.
+
+position_duration = interactive_hyperdrive.hyperdrive_interface.pool_config.position_duration
+rng = np.random.default_rng()  # No seed, we want this to be random every time it is executed
+extra_time = int(np.floor(rng.uniform(low=0, high=position_duration)))
+
+# advances time and mines a block
+current_time = interactive_hyperdrive.hyperdrive_interface.current_pool_state.block_time
+checkpoint_duration = interactive_hyperdrive.hyperdrive_interface.pool_config.checkpoint_duration
+
+starting_checkpoint_time = current_time - current_time % checkpoint_duration
+
+chain.advance_time(position_duration + 30)
+
+# create a checkpoint
+current_time = interactive_hyperdrive.hyperdrive_interface.current_pool_state.block_time
+maturity_checkpoint_time = current_time - current_time % checkpoint_duration
+smart_contract_transact(
+    interactive_hyperdrive.hyperdrive_interface.web3,
+    interactive_hyperdrive.hyperdrive_interface.hyperdrive_contract,
+    signer.agent,
+    "checkpoint",
+    maturity_checkpoint_time,
+)
+chain.advance_time(extra_time)
+maturity_checkpoint = interactive_hyperdrive.hyperdrive_interface.hyperdrive_contract.functions.getCheckpoint(
+    maturity_checkpoint_time
+).call()
+
+# %%
+# close them one at a time, check invariants
+
+SECONDS_PER_YEAR = 60 * 60 * 24 * 365
+
+
+def assertion(condition: bool, message: str = "Assertion failed."):
+    """Simple assertion check.
+
+    Parameters
+    ----------
+    condition : bool
+        condition to check.
+    message : str, optional
+        Error message if condtion fails.
+    """
+    if not condition:
+        print(message)
+
+
+for agent, trade in trade_events:
+    if isinstance(trade, OpenLong):
+        close_long_event = agent.close_long(maturity_time=trade.maturity_time, bonds=trade.bond_amount)
+        # 0.05 would be a 5% fee.
+        flat_fee_percent = interactive_hyperdrive.hyperdrive_interface.pool_config.fees.flat
```

Comment:
Lots of this boilerplate can probably be cleaned up by getting the pool config once and saving it as a variable

Reply:
yup